src/http/server.cr

```diff
@@ -355,7 +355,19 @@ class HTTP::Server
     @sockets.each do |socket|
       spawn do
         until closed?
-          spawn handle_client(socket.accept? || break)
+          io = begin
+            socket.accept?
+          rescue e
+            ::puts "#{socket.class}#accept? raised but it shouldn't"
+            ::puts e
+            nil
+          end
+
+          if io
+            # a non nillable version of io is passed to spawn
+            _io = io
```

Comment:
Shoudn't `io` here be already non-nillable? IIRC `if` branch filters out `Nil`.

Reply:
You recall correctly but using `spawn` affects that rule.

Reply:
`io` is closured, so it always retains all types assigned to it. Basically https://github.com/crystal-lang/crystal/issues/3093

Reply:
The ` || break` before did the trick because the `nil` was transformed to a `NoReturn`.
That stop working in a version were I used `spawn` with block, but `|| break` could return probably since there is no block involved in `spawn`, but it was a bit cryptic.